File: Data Structures & Algorithms/pow-x-n/submission-2.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        
        if n == 0:
            return 1
        
        k = 1
        num = 1

        power = abs(n)

        while power:
            if power % 2 == 1:
                num *= x
            x *= x
            power //= 2
            

        return num if n >= 0 else 1 / num
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> nums_set(nums.begin(), nums.end());        
        int max_streak = 0; 

        for (const auto& num : nums_set) {
            if (!nums_set.contains(num - 1)) {
                int current_num = num;
                int current_streak = 1;

                while (nums_set.contains(current_num + 1)) {
                    current_num += 1;
                    current_streak += 1;
                }

                max_streak = max(max_streak, current_streak);
            }
        }
        
        return max_streak;
    }
};
